### DISCUSSION TITLE
x-on:select does not work

### DISCUSSION THREAD

**Discussion by behvandi11** [wireui/wireui#452] [BUG]
Hi. in x-select component `x-on:select` event does not work. using `x-on:selected` worked but returns past value and needs a debounce to returns current selected result. please hint me to get selected value in alpine. some thing like `x-on:change` in x-native-select

**Comment by Krato** [comment]
For mi is working as expected.

`x-on:selected="console.log($event.detail)"`

Should print in console the current selected value.

Maybe docs should be updated to inform the users how to get the current value

**Reply by behvandi11** [reply]
> For mi is rworking as expected.
> 
> `x-on:selected="console.log($event.detail)"`
> 
> Should print in console the current selected value.
> 
> Maybe docs should be updated to inform the users how to get the current value

Thanks Eric. yes `$event.detail` returns a proxy and by `Object.values($event.details)` could get current value. also an easier way is using:
`x-on:selected.debounce="$event.target.value"`
this is important to use debounce for current val in above example.
I think there can be a more standard way. without use debounce or proxy extract
